tabular/tests/unittests/models/test_advanced.py

```diff
@@ -17,3 +18,44 @@ def test_bagged_predict_children(model_fit_helper):
         check_predict_children=True,
         sample_size=100,
     )
+
+
+def test_resource_constraints(fit_helper, dataset_loader_helper):
+    """
+    Verify that num_cpus and num_gpus are respected when specified in the fit call.
+    Also verifies that constraints are respected for weighted ensemble models and during refit_full.
+    Also verifies that specifying num_cpus > max_cpus will use max_cpus.
```

Comment:
just curious, why only testing this for CPU but not for GPU?

Reply:
We don't have GPU tests for tabular currently. This is something I hope to add in 1.2 release if we find time to dive into better GPU support. I have created #4500 to track as well as added a TODO in the unit test.